lib/puppet/provider/rabbitmq_binding/rabbitmqadmin.rb

```diff
@@ -51,11 +51,13 @@ def self.instances
         end
         if (source_name != '')
           binding = {
+            :source           => source_name,
+            :dest             => destination_name,
+            :vhost            => vhost,
             :destination_type => destination_type,
             :routing_key      => routing_key,
             :arguments        => JSON.parse(arguments),
             :ensure           => :present,
-            :name             => "%s@%s@%s" % [source_name, destination_name, vhost],
```

Comment:
This should be `"%s@%s@%s@%s" % [source_name, destination_name, vhost, routing_key],`